solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/util/JCSMPSessionProducerManager.java

```diff
@@ -39,17 +39,19 @@ public static class CloudStreamEventHandler implements JCSMPStreamingPublishCorr
 
 		@Override
 		public void responseReceivedEx(Object correlationKey) {
-			if (correlationKey instanceof ErrorChannelSendingCorrelationKey) {
-				ErrorChannelSendingCorrelationKey key = (ErrorChannelSendingCorrelationKey) correlationKey;
+			if (correlationKey instanceof BatchProxyCorrelationKey batchProxyCorrelationKey) {
```

Comment:
correlationKey  passed here is eventually not of type BatchProxyCorrelationKey, this is clarification to my comment from above

Reply:
Fixed.